### PR TITLE
Nix inside Docker for easier CI

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: scripts/doc_deploy.sh [hunk 1]
```diff
@@ -43,4 +43,10 @@ fi
 git checkout -q -b ${BRANCH_NAME}
 git add .
 git commit -q -m "Travis update from ${TRAVIS_BUILD_NUMBER}"
+
+if [ "${TRAVIS_PULL_REQUEST}" != "false" -o "${TRAVIS_BRANCH}" != "develop" ]; then
```
Comment (rasendubi) [inline scripts/doc_deploy.sh]: Have you considered Travis' [conditional items](https://docs.travis-ci.com/user/conditional-builds-stages-jobs/) or [script deployment](https://docs.travis-ci.com/user/deployment/script/) with [`on` conditions](https://docs.travis-ci.com/user/deployment#Conditional-Releases-with-on%3A)?

**Comment by rasendubi** [comment]
Do you know Travis [has direct Nix support](https://docs.travis-ci.com/user/languages/nix/)?

File: scripts/doc_deploy.sh (hunk 1 above)
Comment (forGGe) [reply scripts/doc_deploy.sh]: I do want to run `doc_deploy.sh` every time (to make sure changes in deploy script is "tested" too)
What I don't need is that final push. So it easier for now to keep it as shell script with some conditionals,

**Comment by forGGe** [comment]
> Do you know Travis has direct Nix support?

Yes, however, I need docker with Nix inside for #317 . So why not test both docker image and nix in the same run? :)

**Pull request merged**